### PR TITLE
under windows environment  the publicKeyPath and privateKeyPath should replace from \...

### PR DESCRIPTION
**Pull request by cloudtrends**
Under windows environment the publicKeyPath and privateKeyPath should replace from \  to / when send them into injectkeys.py

### CODE REVIEW

File: server/src/com/cloud/server/ConfigurationServerImpl.java [hunk 1]
```diff
@@ -894,6 +894,13 @@ protected void injectSshKeysIntoSystemVmIsoPatch(String publicKeyPath, String pr
         } else {
             command = new Script("/bin/bash", s_logger);
         }
+        String osname = System.getProperty("os.name");
```
Comment (karuturi) [inline server/src/com/cloud/server/ConfigurationServerImpl.java]: There is an isOnWindows() function in this class which can be reused

File: server/src/com/cloud/server/ConfigurationServerImpl.java [hunk 2]
```diff
@@ -894,6 +894,13 @@ protected void injectSshKeysIntoSystemVmIsoPatch(String publicKeyPath, String pr
...
+        if (osname.startsWith("Windows")) {
+            scriptPath = scriptPath.replaceAll("\\\\" ,"/" );
```
Comment (karuturi) [inline server/src/com/cloud/server/ConfigurationServerImpl.java]: testWindowsScript#ConfigurationServerImplTest TestCase might break with this change.

**Comment by karuturi** [comment]
In case of windows, python scripts are executed and I think it runs fine from cygwin. Is this change targeted to make it work from cmd?

File: server/src/com/cloud/server/ConfigurationServerImpl.java (hunk 1 above)
Comment (cloudtrends) [inline server/src/com/cloud/server/ConfigurationServerImpl.java]: Hi  karuturi ,

thanks for your kindly help , yes , I should use the isOnWindows function .

ChunFeng

**Comment by cloudtrends** [comment]
My test environment is on Windows 2008 & cygwin 64 bits version .

I can reoccur the error's by below python scripts:

---------------------------  cygwin_python_test.py -------------------
import os.path
import sys
file_name = sys.argv[1]
if os.path.exists(file_name):
    print file_name
else:
    print "can not find"
-------------------------------- end  --------------------------

test it by , touch a file : F:/sourcecode/python/a-exist.txt

First test with Backslash :  
$ python cygwin_python_test.py  F:\sourcecode\python\a-exist.txt
can not find

Then with slash : 

$ python cygwin_python_test.py  F:/sourcecode/python/a-exist.txt
F:/sourcecode/python/a-exist.txt

------------

So , I thinks we should change the backslash to slash in windows environment .

**Comment by cloudtrends** [comment]
Hi karuturi ,

thanks for point out that I should use isOnWindows function , very appreciation.


ChunFeng

**Comment by karuturi** [comment]
Thanks for the patch. Agree to the slash change. I am pushing this to master(4.6). Let me know if you want it on any other branch.

**Pull request closed without merging**